pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -234,6 +381,7 @@ func (p *Propagator) processEvent() {
 			return
 		}
 		p.removeNode(existingNode)
+		// this doesn't require a RLock, because it's in the same thread that updates the dependents.
```

Comment:
are you saying that the lock is already held by another function on the call stack?